pkg/addonfactory/helm_agentaddon.go

```diff
@@ -88,15 +92,29 @@ func (a *HelmAgentAddon) Manifests(
 			continue
 		}
 		klog.V(4).Infof("rendered template: %v", data)
-		object, _, err := a.decoder.Decode([]byte(data), nil, nil)
-		if err != nil {
-			if runtime.IsMissingKind(err) {
-				klog.V(4).Infof("Skipping template %v, reason: %v", k, err)
-				continue
+
+		yamlReader := yaml.NewYAMLReader(bufio.NewReader(strings.NewReader(data)))
+		for {
+			b, err := yamlReader.Read()
+			if err == io.EOF {
+				break
+			}
+			if err != nil {
+				return nil, err
+			}
+			if len(b) != 0 {
+				object, _, err := a.decoder.Decode(b, nil, nil)
+				if err != nil {
+					if runtime.IsMissingKind(err) {
+						klog.V(4).Infof("Skipping template %v, reason: %v", k, err)
```

Comment:
should we skip or just return err here?

Reply:
vote for "return err", because in cluster-proxy perspective, any resources missing can cause function crash.

Reply:
from this PR https://github.com/open-cluster-management-io/addon-framework/pull/72 described, it's better to stay 'skip', otherwise `policy` addon will be affect.

Reply:
I see, we need to add a note here.

Reply:
Comment added, please take another look.